### ISSUE TITLE
compile error on android 2.2rc

### ISSUE THREAD

**Issue by jjhesk** [toddway/ListOfThings#1]
Configuration on demand is an incubating feature.
Incremental java compilation is an incubating feature.
extractProdDebugAnnotations is incompatible with java 8 sources and has been disabled.
extractProdReleaseAnnotations is incompatible with java 8 sources and has been disabled.
extractDevDebugAnnotations is incompatible with java 8 sources and has been disabled.
extractDevReleaseAnnotations is incompatible with java 8 sources and has been disabled.
Jack is required to support java 8 language features.
Jack is required to support java 8 language features.
Jack is required to support java 8 language features.
Jack is required to support java 8 language features.
:app:preBuild UP-TO-DATE
:app:preDevDebugBuild UP-TO-DATE
:app:checkDevDebugManifest
:app:preDevReleaseBuild UP-TO-DATE
:app:preProdDebugBuild UP-TO-DATE
:app:preProdReleaseBuild UP-TO-DATE
:core:compileJava UP-TO-DATE
:core:processResources UP-TO-DATE
:core:classes UP-TO-DATE
:core:jar UP-TO-DATE
:data:preBuild UP-TO-DATE
:data:preDevDebugBuild UP-TO-DATE
:data:compileDevDebugNdk UP-TO-DATE
:data:compileLint
:data:copyDevDebugLint UP-TO-DATE
:data:checkDevDebugManifest
:data:preDevDebugAndroidTestBuild UP-TO-DATE
:data:preDevDebugUnitTestBuild UP-TO-DATE
:data:preDevReleaseBuild UP-TO-DATE
:data:preDevReleaseUnitTestBuild UP-TO-DATE
:data:preProdDebugAndroidTestBuild UP-TO-DATE
:data:preProdDebugBuild UP-TO-DATE
:data:preProdDebugUnitTestBuild UP-TO-DATE
:data:preProdReleaseBuild UP-TO-DATE
:data:preProdReleaseUnitTestBuild UP-TO-DATE
:data:prepareComAndroidSupportAnimatedVectorDrawable2330Library UP-TO-DATE
:data:prepareComAndroidSupportAppcompatV72330Library UP-TO-DATE
:data:prepareComAndroidSupportSupportV42330Library UP-TO-DATE
:data:prepareComAndroidSupportSupportVectorDrawable2330Library UP-TO-DATE
:data:prepareDevDebugDependencies
:data:compileDevDebugAidl UP-TO-DATE
:data:compileDevDebugRenderscript UP-TO-DATE
:data:generateDevDebugBuildConfig UP-TO-DATE
:data:mergeDevDebugShaders UP-TO-DATE
:data:compileDevDebugShaders UP-TO-DATE
:data:generateDevDebugAssets UP-TO-DATE
:data:mergeDevDebugAssets UP-TO-DATE
:data:generateDevDebugResValues UP-TO-DATE
:data:generateDevDebugResources UP-TO-DATE
:data:mergeDevDebugResources UP-TO-DATE
:data:processDevDebugManifest UP-TO-DATE
:data:processDevDebugResources UP-TO-DATE
:data:generateDevDebugSources UP-TO-DATE
:data:incrementalDevDebugJavaCompilationSafeguard UP-TO-DATE
:data:compileDevDebugJavaWithJavac
An exception has occurred in the compiler (1.8.0_101). Please file a bug against the Java compiler via the Java bug reporting page (http://bugreport.java.com) after checking the Bug Database (http://bugs.java.com) for duplicates. Include your program and the following diagnostic in your report. Thank you.
com.sun.tools.javac.code.Symbol$CompletionFailure: class file for java.lang.invoke.MethodType not found

 FAILED
:data:compileRetrolambdaDevDebug UP-TO-DATE

FAILURE: Build failed with an exception.
- What went wrong:
  Execution failed for task ':data:compileDevDebugJavaWithJavac'.
  
  > Compilation failed; see the compiler error output for details.
- Try:
  Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

**Comment by Winghin2517** [comment]
I'm getting the same error when building on Android Studios 2.2. Please can you let us know how to fix this.

**Comment by waterdaan** [comment]
I had the same issue on Windows 10.
I'm not sure if this is any good, but it works.....

in gradle.properties comment out 
#JAVA7_HOME=/Library/Java/JavaVirtualMachines/jdk1.7.0_79.jdk/Contents/Home
#JAVA8_HOME=/Library/Java/JavaVirtualMachines/jdk1.8.0_92.jdk/Contents/Home

in \app\build.gradle\app  and  \data\build.gradle make these changes
 jdk System.getenv("JAVA8_HOME")
 oldJdk System.getenv("JAVA7_HOME")
 //oldJdk JAVA7_HOME
// jdk JAVA8_HOME

add system variables JAVA8_HOME JAVA7_HOME in windows
https://confluence.atlassian.com/doc/setting-the-java_home-variable-in-windows-8895.html

**Comment by toddway** [comment]
Yeah, this project uses https://github.com/orfjackal/retrolambda so you'll need to be sure JAVA7_HOME and JAVA8_HOME point to the JDKs on your local machine.  Your paths may be different depending on OS and JDK versions.
